website/docs/dev_addon_creation.md

```diff
@@ -275,10 +275,168 @@ If you are curious, here's what happens when you access the api without logging
 
 
 ### Web Action
-:::note TODO
-Write about it and add an example.
+
+Web actions let users trigger AYON Addons CLI actions directly from the AYON server to perform actions on their machines. They run through **shims**, which are automatically set up when users install the AYON launcher version `1.1.0` or higher.
+
+Web actions are added in two steps:
+
+1. **Client-Side CLI Actions**: Develop the addon CLI actions in the client-side code of the addon. More about them in the next section [CLI Interface](dev_addon_creation.md#cli-interface).
+2. **Server-Side Web Actions**: Implement web actions in the server-side code of the addon and link them to their corresponding addon CLI actions.
+
+<!-- TODO: We may expand later on `SimpleActionManifest` and `ActionExecutor`. -->
+
+<details><summary>Simple Web Action Example</summary>
+
+This simple example triggers a QT dialog on the user's machine, showing the folder path from which the action was triggered.
+
+![](assets/addon_dev/web_action_example.png)
+
+```python title="my_addon/client/my_addon/__init__.py"
+from qtpy import QtWidgets
```

Comment:
It's too late, but Qt (or any other runtime dependency) should never be imported in addon...

Reply:
> For reference, should it be imported inside the function like [here](https://github.com/ynput/example-studio-addon/blob/406ac8d56bb7ae3ed556859bd51b2081c79e8a8d/client/my_studio_addon/addon.py#L69)?

Yes, usually by importing Qt it in the function, or by importing file that has Qt implementation inside the function.